### PR TITLE
cache result of fetch_files_from_pr function

### PR DESCRIPTION
**Pull request by boegel**
mostly done to speed up the test suite (in the context of #3482)

### CODE REVIEW

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -377,16 +378,34 @@ def download_repo(repo=GITHUB_EASYCONFIGS_REPO, branch='master', account=GITHUB_
...
+            _log.debug("Using cached value for fetch_files_from_pr for PR #%s (%s)", pr, kwargs)
+            return cache[key]
+        else:
+            res = func(pr, *args, **kwargs)
+            cache[key] = res
+            return res
+
+    # expose clear/update methods of cache to wrapped function
+    cache_aware_func.clear_cache = cache.clear
+    cache_aware_func.update_cache = cache.update
```
Comment (migueldiascosta) [inline easybuild/tools/github.py:403]: just to check if I understand correctly, these two methods are not (yet) used anywhere, right?

File: easybuild/tools/github.py [hunk 2]
```diff
@@ -377,16 +378,34 @@ def download_repo(repo=GITHUB_EASYCONFIGS_REPO, branch='master', account=GITHUB_
...
-    return fetch_files_from_pr(pr, path, github_user, github_repo=GITHUB_EASYBLOCKS_REPO)
+def pr_files_cache(func):
+    """
+    Decorator to cache result of fetch_files_from_pr.
+    """
+    cache = {}
 
+    @functools.wraps(func)
+    def cache_aware_func(pr, *args, **kwargs):
+        """Retrieve cached resul, or fetch files from PR & cache result."""
```
Comment (akesandgren) [inline easybuild/tools/github.py:389]: Typo s/resul/result/

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/tools/github.py (hunk 1 above)
Comment (boegel) [reply easybuild/tools/github.py:403]: Indeed, they're not, but they could come in useful at some point.

In the enhanced tests for example (which are still missing in this PR).

**Review by akesandgren — APPROVED** [review]
LGTM

File: easybuild/tools/github.py (hunk 1 above)
Comment (boegel) [reply easybuild/tools/github.py:405]: fixed, tests are using `clear_cache` and `update_cache` now

**Review by migueldiascosta — APPROVED** [review]
lgtm

**Comment by migueldiascosta** [comment]
Going in, thanks @boegel!

**Pull request merged**